staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -325,65 +333,67 @@ func (o *DebugOptions) Run(f cmdutil.Factory, cmd *cobra.Command) error {
 	}
 	o.podClient = clientset.CoreV1()
 
-	r := f.NewBuilder().
+	builder := f.NewBuilder().
 		WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
 		NamespaceParam(o.Namespace).DefaultNamespace().ResourceNames("pods", o.TargetNames...).
-		Do()
-	if err := r.Err(); err != nil {
-		return err
+		SingleResourceType().
+		Flatten()
+	if len(o.FilenameOptions.Filenames) > 0 {
+		builder.FilenameParam(o.explicitNamespace, &o.FilenameOptions)
```

Comment:
It looks like it's fine to add `FilenameParam` unconditionally, as `kubectl exec` does.

Reply:
Good point, let me try that and assure that it is working in that way

Reply:
I asked in sig-cli https://kubernetes.slack.com/archives/C2GL57FJ4/p1658931680284329